code/modules/mob/dead/observer/freelook/marker/marker_chunk.dm

```diff
@@ -0,0 +1,147 @@
+#define CHUNK_SIZE 16 // Only chunk sizes that are to the power of 2. E.g: 2, 4, 8, 16, etc..
+
+// CAMERA CHUNK
+//
+// A 16x16 grid of the map with a list of turfs that can be seen, are visible and are dimmed.
+// Allows the Marker Eyes to stream these chunks and know what it can and cannot see.
+
+/datum/markerchunk
+	/// Visible turfs in this chunk
+	var/list/visibleTurfs
+	/// Assoc list of the form: list(source = list(turfs source can see in this chunk))
+	/// Contains all vision sources
+	var/list/visionSources
+	/// Vision sources that don't update in hasChanged
+	var/list/rangeVisionSources
+	/// Vision sources that update in hasChanged
+	var/list/viewVisionSources
+	/// List of all turfs = image that masks static
+	var/list/turfs
+	/// List of active images, i.e list of masks for visible turfs
+	var/list/active_masks
+	/// Camera mobs that can see turfs in our chunk
+	var/list/seenby
+	/// Lazy list, In update() things in it will reculculate what turfs they can see
+	var/list/queued_for_update
+
+	var/x = 0
+	var/y = 0
+	var/z = 0
+
+/// Create a new camera chunk, since the chunks are made as they are needed.
+/datum/markerchunk/New(x, y, z, list/netVisionSources, image/example)
+	visibleTurfs = list()
+	visionSources = list()
+	rangeVisionSources = list()
+	viewVisionSources = list()
+	turfs = list()
+	active_masks = list()
+	seenby = list()
+
+	x &= ~(CHUNK_SIZE - 1)
+	y &= ~(CHUNK_SIZE - 1)
+
+	src.x = x
+	src.y = y
+	src.z = z
+
+	for(var/turf/t as anything in block(locate(max(x, 1), max(y, 1), z), locate(min(x + CHUNK_SIZE - 1, world.maxx), min(y + CHUNK_SIZE - 1, world.maxy), z)))
+		turfs[t] = image(example, t)
+
+	var/turf/centre_turf = locate(x + (CHUNK_SIZE / 2), y + (CHUNK_SIZE / 2), z)
+	for(var/atom/source as anything in netVisionSources)
+		if(source.z != z)
+			continue
+		if(!IN_GIVEN_RANGE(source, centre_turf, CHUNK_SIZE))
+			continue
+
+		var/list/visible = list()
+		visionSources[source] = visible
+		for(var/turf/vis_turf as anything in (source.get_visualnet_tiles() & turfs))
+			visible += vis_turf
+			visibleTurfs |= vis_turf
+			active_masks |= turfs[vis_turf]
```

Comment:
Because this is `New()`, wouldn't all these entries be unique anyway? `|=` overhead does start to get pretty big in the hundreds.

Reply:
Since it's inside of a for loop we can't be sure turfs are unique